.gitignore

```diff
@@ -0,0 +1,12 @@
+settings.json
+#Not dirs
+!/**/
+#Not python files
+!*.py
+#And not this file
+!.gitignore
+#Add any other files below:
+.vscode
+.venv
+.python-version
+uv.lock
```

Comment:
we need the uv.lock and .python-version committed since they tell uv what version stuff is

Reply:
no the lock is also necessary for that since the pyproject.toml can have ranges